internal/pkg/provision/providers/docker/create.go

```diff
@@ -35,16 +40,14 @@ func (p *provisioner) Create(ctx context.Context, request provision.ClusterReque
 		return nil, err
 	}
 
-	fmt.Fprintln(options.LogWriter, "generating PKI and tokens")
-
-	input, err := generate.NewInput(request.Name, fmt.Sprintf("https://%s:6443", initNode.IP), request.KubernetesVersion)
-	if err != nil {
-		return nil, err
-	}
+	b64Configs := map[string]string{"Init": "", "ControlPlane": "", "Join": ""}
+	talosConfig := &config.Config{}
 
+	endpoints := []string{"127.0.0.1"}
 	if options.ForceEndpoint != "" {
-		input.AdditionalSubjectAltNames = append(input.AdditionalSubjectAltNames, options.ForceEndpoint)
-		input.AdditionalMachineCertSANs = append(input.AdditionalMachineCertSANs, options.ForceEndpoint)
+		endpoints = []string{options.ForceEndpoint}
+	} else if options.ForceInitNodeAsEndpoint {
```

Comment:
Feels like this whole piece of code which deals with config generation (or reading existing config) can be moved out of the provisioner, as `Create()` function expects `ClusterRequest` which now contains config data (in `NodeRequest`), so Provisioner should get it pre-filled I guess. 

If there's any kind of dependency on config I think we can pass it in a parsed format in `NodeRequest`.